### ISSUE TITLE
Common component to different routes has different behaviour loading JS workers

### ISSUE THREAD

**Issue by frantic0** [roxiness/routify#182]
Hello

I’m using Routify to control internal routing in my application to enable navigation between home `/`,  `/playground`, and dynamic nested subroutes, such as `/tutorial/[chapter]/[section]/`.

Navigation and client-side routing seem to be working fine and components are loading well. However, I found a strange behaviour which seems to be related to navigation.

I have a common component which mounts in different routes ( the `/playground/` and `/tutorial/[chapter]/[section]/` routes). This component loads a pair of JS workers from the public/ folder. Everything works well when the component mounts in the /playground route. 

However, when it mounts on a  `/tutorial/[chapter]/[section]/` route does not load the workers and I get these errors:

Failed to load resource: the server responded with a status of 404 (Not Found) http://localhost:8080/tutorial/01-introduction/02-layout/87ec8fb41b849e935d4c.worker.js
Failed to load resource: the server responded with a status of 404 (Not Found) 
http://localhost:8080/tutorial/01-introduction/02-layout/e93d4586aef9605e4598.worker.js

I’m not sure why the component tries to load the workers from that public folder directory when on this `/tutorial/[chapter]/[section]/` route. It loads correctly in the `/playground` from the root of the public folder directory...

Am I missing something?

Thanks

**Comment by jakobrosenberg** [comment]
Hi @frantic0, off the bat, I'd say you're loading service workers on a relative path.

When using relative paths, your browser looks at the last slash `/` to determine the current folder.
|url                                                                            | folder                 |
|-------------------------------------------------------|-------------------|
|localhost:8080                                                         | /                         |
|localhost:8080/tutorial                                            | /                         |
|localhost:8080/tutorial/                                           | /tutorial/            |
|localhost:8080/tutorial/01-introduction/02-layout/| /tutorial/01-introduction/02-layout/|

To fix this issue, try prefixing the worker url with a `/`.

**Comment by frantic0** [comment]
Hi @jakobrosenberg thanks for hinting that I need to provide absolute URL for all the paths. I had already made the routes absolute by adding the prefix `/` to make sure that all routes loaded correctly, but it led to figuring out that webpack was not generating absolute URLs.

I found that adding the following output.publicPath: "/" to webpack config would sort out that part of the problem and the workers were found and loaded. 

Cheers!
